src/client/Client.cc

```diff
@@ -7956,6 +7956,12 @@ int Client::readlinkat(int dirfd, const char *relpath, char *buf, loff_t size, c
     return r;
   }
 
+  if (!strcmp(relpath, "")) {
+    if (!dirinode.get()->is_symlink())
```

Comment:
So in case if the path is a normal file which would mean dirinode would not be an symlink, wouldn't this? Shouldn't this be without the negation?

Reply:
> So in case if the path is a normal file which would mean dirinode would not be an symlink, wouldn't this mean a bug? Shouldn't this be without the negation?

That's just an invalid case for which we return `ENOENT` here.

Reply:
Oh sorry, I should've just closed this one, this was just an initial screening while taking a look this PR. Then realised we're talking about `readlinkat()` ....